src/index.js

```diff
@@ -1,69 +1,148 @@
 import postcss from 'postcss';
+import topologicalSort from './topologicalSort';
 
-const declWhitelist = ['composes'],
-  declFilter = new RegExp( `^(${declWhitelist.join( '|' )})$` ),
-  matchImports = /^(.+?)\s+from\s+(?:"([^"]+)"|'([^']+)'|(global))$/,
-  icssImport = /^:import\((?:"([^"]+)"|'([^']+)')\)/;
+const declWhitelist = ['composes'];
+const declFilter = new RegExp( `^(${declWhitelist.join( '|' )})$` );
+const matchImports = /^(.+?)\s+from\s+(?:"([^"]+)"|'([^']+)'|(global))$/;
+const icssImport = /^:import\((?:"([^"]+)"|'([^']+)')\)/;
 
-const processor = postcss.plugin( 'modules-extract-imports', function ( options ) {
-  return ( css ) => {
-    let imports = {},
-      importIndex = 0,
-      createImportedName = options && options.createImportedName || (( importName/*, path*/ ) => `i__imported_${importName.replace( /\W/g, '_' )}_${importIndex++}`);
+const VISITED_MARKER = 1;
+
+function createParentName(rule, root) {
+  return `__${root.index(rule.parent)}_${rule.selector}`;
+}
+
+function serializeImports(imports) {
+  return imports.map(importPath => '`' + importPath + '`').join(', ');
+}
+
+function updateGraph(importId, parentId, graph, visited) {
+  const siblingsId = parentId + '_' + 'siblings';
+  const visitedId = parentId + '_' + importId;
+
+  if (visited[visitedId] !== VISITED_MARKER) {
+    if (!Array.isArray(visited[siblingsId])) visited[siblingsId] = [];
+
+    const siblings = visited[siblingsId];
+
+    if (Array.isArray(graph[importId]))
+      graph[importId] = graph[importId].concat(siblings);
+    else
+      graph[importId] = siblings.slice();
+
+    visited[visitedId] = VISITED_MARKER;
+    siblings.push(importId);
+  }
+}
+
+const processor = postcss.plugin('modules-extract-imports', function (options = {}) {
+  const failOnWrongOrder = options.failOnWrongOrder;
```

Comment:
Not sure about the option's name `failOnWrongOrder`. In case you have a better variant — tell me.

The purpose of it is to warn a user if he made a dependency order which isn't possible to resolve properly:

```javascript
.a {
  composes: aa from './aa.css';
  composes: bb from './bb.css';
}

.b {
  composes: bb from './bb.css';
  composes: aa from './aa.css';
}
```